graftm/hmmer.py

```
@@ -598,63 +592,87 @@ def p_search(self, files, args, run_stats, base, unpack, raw_reads):
+        
+        elif search_method == 'diamond':
+            #run diamond
+            search_result =  Diamond(
+                                     database=os.path.join(
+                                                           gpkg._base_directory, \
+                                                           gpkg._contents_hash[gpkg.DIAMOND_DATABASE_KEY]
+                                                           ),
+                                     threads=args.threads,
+                                     evalue=args.eval,
+                                     ).run(
+                                           raw_reads,
+                                           args.input_sequence_type
+                                           )
+            search_result=[search_result]
+            for result in search_result:
+                hit_readnames=[x[0] for x in result.each([SequenceSearchResult.QUERY_ID_FIELD])]
+        else:
+            raise Exception("Programming error: unexpected search_method %s" % search_method)
+        with tempfile.NamedTemporaryFile(prefix='graftm_readnames') as readnames:
+            orfm_regex = re.compile('^(\S+)_(\d+)_(\d)_(\d+)')
+
+            for read in hit_readnames:
+                regex_match = orfm_regex.match(read)
+                if regex_match:
+                    readnames.write(regex_match.groups(0)[0]+'\n')
+                else:
+                    readnames.write(read+'\n')
+            readnames.flush()
+            
+            self._extract_from_raw_reads(
+                                        hit_reads_fasta,
+                                        readnames.name,
+                                        raw_reads,
+                                        unpack.format()
+                                        )
 
         if not hit_readnames:
-            return False, run_stats
+            return None, None, None
```

This happens when there are no search_hmms? If so I would just raise an error rather than trying to carry on